Snowflake.Data/Core/SFStatement.cs

```diff
@@ -680,6 +763,138 @@ CancellationToken cancellationToken
             }
         }
 
+        /// <summary>
+        /// Creates a request to get the query status based on query ID.
+        /// </summary>
+        /// <param name="queryId"></param>
+        /// <returns>The request to get the query status.</returns>
+        private SFRestRequest BuildQueryStatusRequest(string queryId)
+        {
+            var queryUri = SfSession.BuildUri(RestPath.SF_MONITOR_QUERY_PATH + queryId);
+
+            return new SFRestRequest
+            {
+                Url = queryUri,
+                authorizationToken = string.Format(SF_AUTHORIZATION_SNOWFLAKE_FMT, SfSession.sessionToken),
+                serviceName = SfSession.ParameterMap.ContainsKey(SFSessionParameter.SERVICE_NAME)
+                                ? (String)SfSession.ParameterMap[SFSessionParameter.SERVICE_NAME] : null,
+                HttpTimeout = Timeout.InfiniteTimeSpan,
+                RestTimeout = Timeout.InfiniteTimeSpan,

```

Comment:
I'm just thinking if infinitive waiting for getting the query status is good... But from the other hand we do the same when executing a normal query.